noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/components/private_call_data_validator.nr

```diff
@@ -37,22 +35,28 @@ unconstrained fn find_first_revertible_private_call_request_index(public_inputs:
     )
 }
 
-fn validate_caller_context(caller_context: CallerContext, this_context: CallContext) {
-    let matching_caller_context = caller_context.msg_sender.eq(this_context.msg_sender)
-        & caller_context.storage_contract_address.eq(this_context.storage_contract_address);
-    let hidden_caller_context = caller_context.is_hidden();
-    assert(matching_caller_context | hidden_caller_context, "invalid caller context");
-
-    assert_eq(caller_context.is_static_call, this_context.is_static_call, "mismatch is_static_call flag");
-}
-
-fn validate_call_request(request: CallRequest, hash: Field, caller: PrivateCallStackItem) {
-    if hash != 0 {
-        assert_eq(request.hash, hash, "call stack hash does not match call request hash");
-        assert_eq(request.caller_contract_address, caller.contract_address, "invalid caller contract address");
-        validate_caller_context(request.caller_context, caller.public_inputs.call_context)
+fn validate_call_context(
```

Comment:
This is checked on all the private call requests emitted from a private call.
We've replaced the caller context with call context in the call request, and check the addresses in the call context instead.
Which means we don't have to check the addresses again in a nested call (in `validate_against_call_request`), it's legit as long as the call context in the request matches the one in the call stack item.